packages/js/data/src/setting-options/types.ts

```diff
@@ -0,0 +1,160 @@
+/**
+ * Represents the value of a setting. It's unknown because any value can be stored.
+ */
+export type SettingValue = unknown;
```

Comment:
Gut feel that having it as unknown might result in a bunch of type overrides but I'm happy to leave it alone for now and see if it proves true

Reply:
It appears that the majority of settings values are strings, although `wc_admin` utilizes `string[]` for a few. Additionally, it seems that any type of value can be registered so I left it as unknown for now.